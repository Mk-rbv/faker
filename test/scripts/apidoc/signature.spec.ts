import { beforeAll, describe, expect, it } from 'vitest';
import { analyzeSignature } from '../../../scripts/apidoc/processing/signature';
import { initMarkdownRenderer } from '../../../scripts/apidoc/utils/markdown';
import { SignatureTest } from './signature.example';
import { loadExampleMethods } from './utils';

beforeAll(initMarkdownRenderer);
const methods = await loadExampleMethods();

describe('signature', () => {
  describe('analyzeSignature()', () => {
    it('dummy dependency to rerun the test if the example changes', () => {
      expect(new SignatureTest()).toBeTruthy();
    });

    it('expected and actual methods are equal', () => {
      expect(Object.keys(methods)).toMatchSnapshot();
    });

    it.each(Object.entries(methods))('%s', async (name, signature) => {
      const actual = await analyzeSignature(signature, '', name);

      expect(actual).toMatchSnapshot();
    });
  });
});
