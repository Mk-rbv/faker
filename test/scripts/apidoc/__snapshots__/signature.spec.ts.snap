// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`signature > analyzeSignature() > expected and actual methods are equal 1`] = `
[
  "noParamMethod",
  "requiredNumberParamMethod",
  "optionalStringParamMethod",
  "defaultBooleanParamMethod",
  "multiParamMethod",
  "functionParamMethod",
  "stringUnionParamMethod",
  "literalUnionParamMethod",
  "recordParamMethod",
  "optionsParamMethod",
  "optionsInlineParamMethodWithDefaults",
  "optionsTypeParamMethodWithDefaults",
  "optionsInterfaceParamMethodWithDefaults",
  "methodWithExample",
  "methodWithDeprecated",
  "methodWithThrows",
  "methodWithMultipleThrows",
  "methodWithDeprecatedOption",
  "methodWithMultipleSeeMarkers",
  "methodWithMultipleSeeMarkersAndBackticks",
  "methodWithSinceMarker",
  "complexArrayParameter",
]
`;

exports[`signature > analyzeSignature() > methodWithSinceMarker 1`] = `
{
  "name": "methodWithSinceMarker",
  "signatures": [
    {
      "deprecated": undefined,
      "description": "Test with since marker.",
      "examples": [],
      "parameters": [],
      "returns": {
        "text": "number",
        "type": "simple",
      },
      "seeAlsos": [],
      "signature": "function methodWithSinceMarker(): number;",
      "since": "1.0.0",
      "throws": [],
    },
  ],
  "sourcePath": "test/scripts/apidoc/signature.example.ts:364:3",
}
`;
